File: frontend/src/types/promoCode.ts
export type DiscountType = 'fixed' | 'percent'

export interface PromoCode {
  id: number
  code: string
  minimumPurchase: number
  discountType: DiscountType
  discountValue: number
  courseIds: number[]
  createdAt: string
  updatedAt: string
}

export interface PromoCodePayload {
  code: string
  minimumPurchase: number
  discountType: DiscountType
  discountValue: number
  courseIds: number[]
}

File: backend/src/main/java/com/courseflow/promocode/PromoCode.java
package com.courseflow.promocode;

import java.math.BigDecimal;
import java.time.OffsetDateTime;
import java.util.List;

public record PromoCode(
    Long id,
    String code,
    BigDecimal minimumPurchase,
    String discountType,
    BigDecimal discountValue,
    List<Long> courseIds,
    OffsetDateTime createdAt,
    OffsetDateTime updatedAt
) {
}
